sdk/storage/azfile/sas/service.go

```diff
@@ -0,0 +1,228 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package sas
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azfile/internal/exported"
+)
+
+// FileSignatureValues is used to generate a Shared Access Signature (SAS) for an Azure Storage file or share.
+// For more information on creating service sas, see https://docs.microsoft.com/rest/api/storageservices/constructing-a-service-sas
+// User Delegation SAS not supported for files service
+type FileSignatureValues struct {
+	Version            string    `param:"sv"`  // If not specified, this defaults to Version
+	Protocol           Protocol  `param:"spr"` // See the Protocol* constants
+	StartTime          time.Time `param:"st"`  // Not specified if IsZero
+	ExpiryTime         time.Time `param:"se"`  // Not specified if IsZero
+	SnapshotTime       time.Time
+	Permissions        string  `param:"sp"` // Create by initializing a SharePermissions or FilePermissions and then call String()
+	IPRange            IPRange `param:"sip"`
+	Identifier         string  `param:"si"`
+	ShareName          string
+	FilePath           string // Ex: "directory/FileName" or "FileName". Use "" to create a Share SAS.
```

Comment:
Do directories in file share have SAS too? if so would path be more generic a name to cover that case too?

Reply:
also if they dont support it now, maybe they might in the future, so might be good to think about how to future proof this?

Reply:
I think SAS is supported for directories as well. I checked the .NET SDK and they do have method to generate SAS in the directory client. What we can do is rename this parameter to `DirectoryOrFilePath`. Thoughts?